### ISSUE TITLE
file_system_client.get_paths only retrieving first 5000 entries

### ISSUE THREAD

**Issue by GBBBAS** [Azure/azure-sdk-for-python#16168]
- **Package Name**: azure-storage-file-datalake
- **Package Version**: 12.2.1
- **Operating System**: linux
- **Python Version**: 3.7.9

**Describe the bug**
Since upgrading to 12.2.1, the file_system_client method get_paths appears to stop after 5000 records. Downgrading the package to 12.2.0 retrieves all records

**To Reproduce**
1. pip install azure-storage-file-datalake==12.2.1
2. Setup file_system client where there are more than 5000 directories and files
```
credential = ClientSecretCredential(directoryId, clientId, clientSecret)
service_client = DataLakeServiceClient(account_url="{}://{}.dfs.core.windows.net".format("https", storageAccount), credential=credential)       
file_system_client = service_client.get_file_system_client(file_system="testfilesystem")
```
3. Get paths 
```
paths = file_system_client.get_paths(path='/'), recursive=True)
get_all_paths = [path.name for path in paths]
```
4. Only 5000 records are retrieved

**Expected behavior**
All records are retrieved, as per beahaviour of version 12.2.0.

**Comment by v-xuto** [comment]
I refer to this test method [test_list_paths_with_max_per_page](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-file-datalake/tests/test_file_system.py#L241) in python sdk test (test_file_system.py). I uploaded 1~9 txt files in my filesystem "testfilesystem".But in different versions, there are different results, I think the results in 12.2.0 version are expected results. 

**1.Package Name and Version: `azure-storage-file-datalake==12.2.0`.** 
Where does generator1 get the path, there will be a continuationation_token, and then generator2 continues to get path after generator1. Picture of output result:
![image](https://user-images.githubusercontent.com/57166602/104982883-8e819680-5a46-11eb-8aa2-202e4ba530f8.png)

**2).Package Name and Version: `azure-storage-file-datalake==12.2.1`.** 
Because the continuationation_token is None, generator2 still starts to get the path from the first file. Picture of output result:
![image](https://user-images.githubusercontent.com/57166602/104982964-bd980800-5a46-11eb-81bc-624f24dbd4a6.png)


**The detailed code I tested is as follows (refer to [test](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-file-datalake/tests/test_file_system.py#L241)):**
```
service_client = DataLakeServiceClient(account_url="{}://{}.dfs.core.windows.net".format("https", storageAccount), credential=credential)       
file_system_client = service_client.get_file_system_client(file_system="testfilesystem")

generator1 = file_system_client.get_paths(upn=True,max_results=6).by_page()
paths1 = list(next(generator1))
print("----First page file names----")
for id, content in enumerate(paths1):
    print(content.name)
print("Summary1 : {}".format(id+1))

print("=============================================================")
generator2 = file_system_client.get_paths(upn=True, max_results=3).by_page(continuation_token=generator1.continuation_token)
paths2 = list(next(generator2))
print("----Second page file names----")
for idx, content in enumerate(paths2):
    print(content.name)
print("Summary2 : {}".format(idx+1))
```

**Comment by EthanNguyen132** [comment]
We're having the exact problem, downgrading to 12.2.0 works for now...

**Comment by SindreEquinor** [comment]
Seeing the same issue

**Comment by kwura** [comment]
Seeing the same issue will try to downgrade and see if that works

**Comment by tasherif-msft** [comment]
Hi @GBBBAS This issue has been resolved in >12.2.3 datalake
